src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -93,28 +94,43 @@ public JsonArray(Buffer buf) {
   /**
    * Get the String at position {@code pos} in the array,
    *
-   * @param pos  the position in the array
-   * @return  the String, or null if a null value present
+   * @param pos the position in the array
+   * @return the String, or null if a null value present
    * @throws java.lang.ClassCastException if the value cannot be converted to String
    */
   public String getString(int pos) {
-    CharSequence cs = (CharSequence)list.get(pos);
-    return cs == null ? null : cs.toString();
+    Object val = list.get(pos);
+
+    if (val == null) {
+      return null;
+    }
+
+    if (val instanceof CharSequence) {
+      return val.toString();
+    } else if (val instanceof Instant) {
+      return ISO_INSTANT.format((Instant) val);
+    } else if (val instanceof byte[]) {
+      return BASE64_ENCODER.encodeToString((byte[]) val);
+    } else if (val instanceof Enum) {
+      return ((Enum) val).name();
+    }
+
+    throw new ClassCastException("class " + val.getClass().getName() + " cannot be cast to class java.lang.String");
```

Comment:
how about instead using `toString()` ?

Reply:
That is also a possibility and if I'm not mistaken how javascript does it but it will change the behavior from the older version that would throw a CCE if the value isn't a CharSequence. I'm open to it, if you prefer

Reply:
I believe that being lenient here is good specially if we use this for simple object mapping with SQL client as the intent is to pass a Map<String, Object> from JsonObject to wrap the row.